packages/client/src/links/httpBatchLink/streamingHttpUtils.ts

```diff
@@ -0,0 +1,170 @@
+// Stream parsing adapted from https://www.loginradius.com/blog/engineering/guest-post/http-streaming-with-nodejs-and-fetch-api/
+
+import { TRPCResponse } from '@trpc/server/rpc';
+import { TRPCClientError } from '../../TRPCClientError';
+import {
+  HTTPBaseRequestOptions,
+  HTTPResult,
+  fetchHTTPResponse,
+  getBody,
+  getUrl,
+} from '../internals/httpUtils';
+import { HTTPHeaders } from '../types';
+
+export async function parseJsonStream<TReturn>(opts: {
+  /**
+   * As given by `(await fetch(url)).body`
+   */
+  readableStream: ReadableStream<Uint8Array> | NodeJS.ReadableStream;
+  /**
+   * Called for each line of the stream
+   */
+  onSingle: (index: number, res: TReturn) => void;
+  /**
+   * Called when the stream is finished
+   */
+  parse?: (text: string) => TReturn;
+  signal?: AbortSignal;
+}): Promise<TReturn | undefined> {
+  const parse = opts.parse ?? JSON.parse;
+
+  let isFirstLine = true;
+  let isFullSink = false;
+  let fullAccumulator = '';
+  const onLine = (line: string) => {
+    if (opts.signal?.aborted) return;
+    if (isFirstLine) {
+      isFirstLine = false;
+      if (line !== '{') {
+        // unexpected format, don't try to parse it as a stream
+        isFullSink = true;
+        fullAccumulator = line;
+      }
```

Comment:
My worry was that `transfer-encoding: chunked` is specced as a hop-by-hop header, meaning any intermediary (cache layer for example) can remove it, but the shape of the string body is still "not an array but an object with index keys".

But we can also consider that users are responsible for their own infra and if theirs doesn't support streaming, don't use that link.

Reply:
If people start using the streaming link and get issues with that sort of stuff, they'll report it.

On your line of thinking, we should also support and valid JSON, since that intermediate layer can store the JSON in another format that then gets printed as 1 line or prettified